bin/fxa-oauth.js

```
@@ -102,10 +154,132 @@ program
+  .action(function() {
+    withTempToken(function(client) {
+      /*jshint camelcase:false*/
+      return client.listClients()
+        .done(function(clients) {
+          p(JSON.stringify(clients.clients, null, 2));
+        }, onError);
+    });
+  });
+
+function promptClient() {
+  var results = {};
+  /*jshint camelcase:false*/
+  var schema = {
+    name: {},
+    redirect_uri: {},
+    image_uri: {},
+    whitelisted: {
+      default: 'true'
+    },
+    can_grant: {
+      prompt: 'Implicit grant permission?',
+      default: 'false'
+    }
+  };
+  var prompts = Object.keys(schema);
+
+  function prompt(name) {
+    var opts = schema[name];
+    if (!opts.prompt) {
+      opts.prompt = name + ':';
+    }
+    return read(opts).spread(function(input) {
+      if (input) {
+        results[name] = input;
+      }
+    });
+  }
+
+  return P.map(prompts, prompt, { concurrency: 1 }).then(function() {
+    return results;
+  });
+}
+
+program
+  .command('register')
+  .description('Register a new OAuth client.')
+  .action(function() {
+    p('Fill in client details...');
+    return promptClient().then(function(client) {
+      log.verbose('register', 'client', client);
+      p('Registering client:', JSON.stringify(client, null, 2));
+      p('');
+      return read({
+        prompt: 'Is this correct?',
+        default: 'no'
+      }).spread(function(answer) {
+        if (toBoolean(answer)) {
+          return withTempToken(function(oauth) {
+            return oauth.registerClient(client);
+          });
+        }
+      });
+    }).done(function(client) {
+      log.verbose('register', 'complete', client);
+      p('Client registered:', JSON.stringify(client, null, 2));
+    }, onError);
+  });
+
+program
+  .command('update <clientId> <property> <value>')
+  .description('Update a property of a client.')
+  .action(function(id, prop, value) {
+    //TODO: validate id,prop,value
+    withTempToken(function(oauth) {
+      var client = { id: id };
```

The oauth server doesn't take "id" in the request body when updating a client record.  This used to be silently ignored but it's now failing.  We need to either accept a matching "id" in the request body on the server, or stop sending it in the client.